src/storage/in_mem_storage_structure/in_mem_column_chunk.cpp

```diff
@@ -127,5 +145,139 @@ void InMemColumnChunk::setValueFromString<common::timestamp_t>(
     copyValue(pageIdx, posInPage, (uint8_t*)&val);
 }
 
+InMemStructColumnChunk::InMemStructColumnChunk(
+    common::DataType dataType, common::offset_t startOffset, common::offset_t endOffset) {
+    this->dataType = std::move(dataType);
+    auto childrenTypes =
+        reinterpret_cast<common::StructTypeInfo*>(this->dataType.getExtraTypeInfo())
+            ->getChildrenTypes();
+    for (auto& childType : childrenTypes) {
+        inMemColumnChunksForFields.push_back(InMemColumnChunkFactory::getInMemColumnChunk(
+            *childType, startOffset, endOffset, common::Types::getDataTypeSize(*childType),
+            common::BufferPoolConstants::PAGE_4KB_SIZE /
+                common::Types::getDataTypeSize(*childType)));
+    }
+}
+
+void InMemStructColumnChunk::copyStructValueToFields(arrow::Array& array, uint64_t posInArray,
+    const common::CopyDescription& copyDescription, common::offset_t startOffset,
+    uint64_t numValues) {
+    // TODO(Ziyi): support null values in struct.
+    for (auto i = 0u; i < numValues; i++) {
+        auto& stringArray = (arrow::StringArray&)array;
+        auto structView = stringArray.GetView(i + posInArray);
+        auto structTypeInfo =
+            reinterpret_cast<common::StructTypeInfo*>(dataType.getExtraTypeInfo());
+        auto structFieldTypes = structTypeInfo->getChildrenTypes();
+        auto structFieldNames = structTypeInfo->getChildrenNames();
+        std::regex whiteSpacePattern{"\\s"};
+        // Removes the leading and trailing '{', '}';
+        auto structString =
+            std::string(structView.data(), structView.length()).substr(1, structView.length() - 2);
+        std::istringstream structStream{structString};
+        std::string structField;
+        while (getline(structStream, structField, ',')) {
```

Comment:
Let's see if you can avoid using `istringstream` and create new string here.

Reply:
Replaced it with `StringUtils::split`